pkg/client/cloudevents/grpcsource/client.go

```diff
@@ -0,0 +1,88 @@
+package grpcsource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift-online/maestro/pkg/api/openapi"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	workv1client "open-cluster-management.io/api/client/work/clientset/versioned/typed/work/v1"
+	workv1 "open-cluster-management.io/api/work/v1"
+
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/generic"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/generic/options/grpc"
+	sourceclient "open-cluster-management.io/sdk-go/pkg/cloudevents/work/source/client"
+	sourcecodec "open-cluster-management.io/sdk-go/pkg/cloudevents/work/source/codec"
+)
+
+func NewMaestroGRPCSourceWorkClient(
+	ctx context.Context,
+	apiClient *openapi.APIClient,
+	opts *grpc.GRPCOptions,
+	sourceID string,
+) (workv1client.WorkV1Interface, error) {
+	if len(sourceID) == 0 {
+		return nil, fmt.Errorf("source id is required")
+	}
+
+	options, err := generic.BuildCloudEventsSourceOptions(opts, fmt.Sprintf("%s-maestro", sourceID), sourceID)
```

Comment:
The `fmt.Sprintf("%s-maestro", sourceID)` is the sourceClientID? How is it used and why is it suffixed by `-maestro`?

Reply:
the `sourceClientID` is not used in gRPC case, it just an interface required (because the mqtt case will need a client ID)

Reply:
Thanks for confirming.